products_application/src/App.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { Component } from 'react';
+import styles from './App.module.scss';
+import { Cards } from './components/Cards/Cards';
+import { Loader } from './components/Loader/Loader';
+import { ErrorButton } from './components/ErrorButton/ErrorButton';
+import { IAppState, IRequestResult } from './types';
+
+export class App extends Component<object, IAppState, string> {
+  state = {
+    inputValue: this.setInputValue(),
+    cards: [],
+    isLoading: true,
+    isNothingFound: false,
+    isButtonDisabled: false,
+  };
+
+  setInputValue() {
+    const keyWord = localStorage.getItem('alyona8891_keyword');
+    if (keyWord) {
+      return keyWord;
+    } else {
+      return '';
+    }
+  }
+
+  componentDidMount(): void {
+    const keyWord = localStorage.getItem('alyona8891_keyword');
+    if (keyWord) {
+      this.setState({
+        isButtonDisabled: true,
+      });
+      fetch(`https://dummyjson.com/products/search?q=${keyWord}`)
```

Comment:
recommend you to put fetching products to separate function to avoid code duplication and get rid of  large `if else` condition in component.
like `fetchProducts(keyWord?: string) { ...return something... }`